tests/cephfs/cephfs_nfs/test_cephfs_upgrade_nfs_standalone_to_nfs_ha_cluster.py

```diff
@@ -0,0 +1,173 @@
+        )
+        json.loads(out)
+        mounting_dir = "".join(
+            random.choice(string.ascii_lowercase + string.digits)
+            for _ in list(range(10))
+        )
+        nfs_mounting_dir = f"/mnt/cephfs_nfs{mounting_dir}_1/"
+        client1.exec_command(sudo=True, cmd=f"mkdir -p {nfs_mounting_dir}")
+
+        command = f"mount -t nfs -o port={port} {nfs_servers[0].node.hostname}:{nfs_export_name} {nfs_mounting_dir}"
+        client1.exec_command(sudo=True, cmd=command, check_ec=False)
+
+        dir_name = "smallfile_dir"
+        client1.exec_command(sudo=True, cmd=f"mkdir -p {nfs_mounting_dir}{dir_name}")
+        smallfile(clients[0], nfs_mounting_dir, dir_name)
+
+        client1.exec_command(sudo=True, cmd=f"echo '{ingress_conf}' > ingress.yaml")
+        client1.exec_command(sudo=True, cmd="ceph orch apply -i ingress.yaml")
+        log.info("validate the Ingress services")
+        fs_util_v1.validate_services(client1, f"ingress.nfs.{nfs_name}")
+
+        log.info("Unmount and remount using Virtual IP")
+        client1.exec_command(
+            sudo=True, cmd=f"umount -l {nfs_mounting_dir}", check_ec=False
+        )
+        command = f"mount -t nfs -o port={port} {virtual_ip}:{nfs_export_name} {nfs_mounting_dir}"
+        client1.exec_command(sudo=True, cmd=command, check_ec=False)
+        log.info(
+            f"Validate if the {dir_name} exists, delete the dir and rerun IOs on new mount path"
+        )
+        validate_nfs_ha_server(client1, nfs_mounting_dir, dir_name)
+
+        log.info("Test completed successfully")
+        return 0
+    except Exception as e:
+        log.error(e)
+        log.error(traceback.format_exc())
+        return 1
+    finally:
+        log.info("Cleaning Up")
+        client1.exec_command(sudo=True, cmd=f"rm -rf {nfs_mounting_dir}*")
+        log.info("Unmount NFS export")
+        client1.exec_command(
+            sudo=True, cmd=f"umount -l {nfs_mounting_dir}", check_ec=False
+        )
+        log.info("Removing the Export")
+        client1.exec_command(
+            sudo=True,
+            cmd=f"ceph nfs export delete {nfs_name} {nfs_export_name}",
+            check_ec=False,
+        )
+        log.info("Removing NFS Cluster")
+        client1.exec_command(
+            sudo=True, cmd=f"ceph nfs cluster rm {nfs_name}", check_ec=False
+        )
+
+
+def smallfile(client, mounting_dir, dir_name):
+    client.exec_command(
+        sudo=True,
+        cmd=f"python3 /home/cephuser/smallfile/smallfile_cli.py --operation create --threads 8 --file-size 10240 "
+        f"--files 10 --top {mounting_dir}{dir_name}",
+    )
+
+
+@retry(CommandFailed, tries=5, delay=5)
+def validate_nfs_ha_server(client, nfs_mounting_dir, dir_name):
+    try:
+        client.exec_command(sudo=True, cmd=f"ls -l {nfs_mounting_dir}{dir_name}")
+        client.exec_command(sudo=True, cmd=f"rm -rf {nfs_mounting_dir}{dir_name}")
```

Comment:
Can we write IO to same directory post upgrade and validate? Is it necessary to delete and create which becomes new altogether.

Reply:
Just to validate the existing folder remains in HA setup, I am doing "ls" and removing an existing folder